src/main/java/emu/grasscutter/game/player/Player.java

```diff
@@ -968,6 +970,18 @@ public List<ShowAvatarInfoOuterClass.ShowAvatarInfo> getShowAvatarInfoList() {
         return showAvatarInfoList;
     }
 
+    public List<Integer> getShowNameCardInfoList() {
+        Player player;
+        if (this.isOnline()) {
+            player = this;
+        } else {
+            player = DatabaseHelper.getPlayerByUid(id);
+        }
```

Comment:
This is pretty much a reimplementation of the existing pattern:

```suggestion
        Player player = Grasscutter.getGameServer().getPlayerByUid(id, true);
```

Though as a method of `Player`, I'd expect that it should already be an offline player object if it ever gets to the point of fetching the offline player object, so it probably doesn't make sense to do the lookup at all.